### PR TITLE
Introduce `gardener` role

### PR DESCRIPTION
**Pull request by Gerrit91**
Consists of multiple PRs:
- https://github.com/metal-stack/releases/pull/106
- https://github.com/metal-stack/metal-roles/pull/143
- https://github.com/metal-stack/mini-lab/pull/135

### CODE REVIEW

**Comment by Gerrit91** [comment]
## TODO

- [OPERATOR] The DisableDNSProviderManagement feature gate has been promoted to beta and is now enabled by default. If you are using the Gardener extension shoot-dns-service make sure to deploy version >= v1.20.0 and to set providerConfig.values.dnsProviderManagement.enabled=true in its controller deployment. The shoot DNS service admission controller (gardener-extension-admission-shoot-dns-service) must be deployed on the garden cluster. 

- Admission controllers for
  - [x] GEPM
  - [ ] GCP
  - [ ] os-metal-extension
  - [ ] networking-calico
  - [ ] networking-cilium
  - [ ] shoot-cert-service
  - [ ] shoot-dns-service

File: control-plane/roles/gardener/defaults/main/main.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+---
+gardener_image_vector_overwrite:
+gardener_component_image_vector_overwrite:
+
+gardener_extension_provider_metal_repo_ref: "{{ gardener_extension_provider_metal_image_tag }}"
+gardener_os_controller_repo_ref: "{{ gardener_os_controller_image_tag }}"
+# TODO: as soon as our changes on cilium controller are merged to upstream, we can remove the path to the fork
+gardener_networking_cilium_repo_ref: "metal-stack/gardener-extension-networking-cilium/{{ gardener_networking_cilium_image_tag }}"
```
Comment (vknabel) [inline control-plane/roles/gardener/defaults/main/main.yaml:8]: Which changes? Is there a PR we could link?

**Review by vknabel — COMMENTED** [review]
Cannot test, but convention wise it looks fine.

File: control-plane/roles/gardener/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# gardener
+
+Deploys Gardener into a virtual garden along with a dedicated ETCD and a set of extension controllers.
+
+Check out the Gardener project for further documentation: https://gardener.cloud/.
+
+## Variables
+
+TODO
```
Comment (vknabel) [inline control-plane/roles/gardener/README.md:9]: TODO

File: control-plane/roles/gardener/defaults/main/main.yaml (hunk 1 above)
Comment (Gerrit91) [reply control-plane/roles/gardener/defaults/main/main.yaml]: I improved the details here a bit when we can remove this override.

File: control-plane/roles/gardener/README.md (hunk 2 above)
Comment (Gerrit91) [reply control-plane/roles/gardener/README.md]: Done.

**Review by vknabel — APPROVED** [review]

**Pull request merged**